documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/util/JCRDocumentsUtil.java

```diff
@@ -485,7 +490,32 @@ public static void retrieveFileContentProperties(Node content, FileNode fileNode
     }
     if (content.hasProperty(NodeTypeConstants.JCR_DATA)) {
       fileNode.setSize(content.getProperty(NodeTypeConstants.JCR_DATA).getLength());
+      fileNode.setSizeWithVersions(computeVersionsSize(content.getParent()));
+
+    }
+  }
+
+  public static long computeVersionsSize(Node node) {
+    long versionSize=0;
+    try {
+      if (node.isNodeType(MIX_VERSIONABLE)) {
+        VersionHistory versionHistory = node.getVersionHistory();
+        VersionIterator iterator = versionHistory.getAllVersions();
+        while (iterator.hasNext()) {
+          Version version = iterator.nextVersion();
+          if (version.hasNode("jcr:frozenNode")) {
+            Node frozen= version.getNode("jcr:frozenNode");
+            if (frozen.hasNode("jcr:content")) {
+              long currentVersionSize = frozen.getNode("jcr:content").getProperty("jcr:data").getLength();
```

Comment:
Could you add a try/catch block here to compute the size of versions if if there is an issue with JCR:CONTENT subnode, sometimes it is corrupt or missing

Reply:
done